pandas/tests/reshape/merge/test_multi.py

```diff
@@ -460,6 +460,30 @@ def test_merge_na_keys(self):
 
         tm.assert_frame_equal(result, expected)
 
+    def test_merge_datetime_index_empty_df(self):
+        data = [
+            [pd.Timestamp("1950-01-01"), "A", 1.5],
+            [pd.Timestamp("1950-01-02"), "B", 1.5],
+        ]
+
+        frame = DataFrame(data, columns=["date", "panel", "data"]).set_index(
+            ["date", "panel"]
+        )
+        other = DataFrame(columns=["date", "panel", "state"]).set_index(
+            ["date", "panel"]
+        )
+        expected_data = [
+            [pd.Timestamp("1950-01-01"), "A", 1.5, pd.NA],
+            [pd.Timestamp("1950-01-02"), "B", 1.5, pd.NA],
+        ]
```

Comment:
Hmm, I would probably use np.nan here instead of pd.NA. Note that the merge actually produces nan but the assertion still passes (arguably a bug in the assert function):
```python
[ins] In [11]: result
Out[11]: 
                  data state
date       panel            
1950-01-01 A       1.5   NaN
1950-01-02 B       1.5   NaN

[ins] In [12]: expected
Out[12]: 
                  data state
date       panel            
1950-01-01 A       1.5  <NA>
1950-01-02 B       1.5  <NA>

[ins] In [13]: tm.assert_frame_equal(result, expected)

[ins] In [14]: 
```

Reply:
The problem with setting it as `np.nan` is the type automatically is inferred as a float64 instead of "object" 
```
>       tm.assert_frame_equal(result, expected)
E       AssertionError: Attributes of DataFrame.iloc[:, 1] (column name="state") are different
E       
E       Attribute "dtype" are different
E       [left]:  object
E       [right]: float64
```

Reply:
You can specify dtypes in the constructor instead of relying on dtype inference, or pass an object array directly

Reply:
👍